storage/innobase/handler/ha_innodb.cc

```diff
@@ -19307,6 +19307,12 @@ static MYSQL_SYSVAR_BOOL(optimize_fulltext_only, innodb_optimize_fulltext_only,
   "Only optimize the Fulltext index of the table",
   NULL, NULL, FALSE);
 
+static MYSQL_SYSVAR_BOOL(sys_tablespace_shrink_extensive,
+  srv_sys_shrink_extensive,
+  PLUGIN_VAR_OPCMDARG,
+  "Shrink the system tablespace by doing extensive check",
+  NULL, NULL, TRUE);
```

Comment:
Instead of repeating the word `extensive`, can we try to explain the effect of setting this option? Such as: "Mark any pages in the system tablespace free that do not belong to any table that resides in the system tablespace. Has no effect if innodb_undo_tablespaces<2."

This option should really be disabled by default, in order to avoid changing any default behaviour after a GA release.